modules/framework/galasa-parent/dev.galasa.framework.api.users/src/main/java/dev/galasa/framework/api/users/internal/routes/UsersDeleteRoute.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.users.internal.routes;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import static dev.galasa.framework.api.common.ServletErrorMessage.*;
+
+import dev.galasa.framework.spi.auth.IInternalAuthToken;
+
+import dev.galasa.framework.api.common.BaseRoute;
+import dev.galasa.framework.api.common.Environment;
+import dev.galasa.framework.api.common.EnvironmentVariables;
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.QueryParameters;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.auth.spi.IAuthService;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.auth.AuthStoreException;
+import dev.galasa.framework.spi.auth.IAuthStoreService;
+import dev.galasa.framework.spi.auth.IUser;
+
+public class UsersDeleteRoute extends BaseRoute{
+
+    // Regex to match endpoint /users/{someLoginId}
+    private static final String path = "\\/([a-zA-Z0-9\\-\\_]+)\\/?";
+
+    private Environment env;
+    private IAuthStoreService authStoreService;
+    private IAuthService authService;
+    private BeanTransformer beanTransformer;
+    private Pattern pathPattern;
+
+    public UsersDeleteRoute(ResponseBuilder responseBuilder, Environment env,
+            IAuthService authService) {
+        super(responseBuilder, path);
+        this.env = env;
+        this.authService = authService;
+        
+        String baseServletUrl = env.getenv(EnvironmentVariables.GALASA_EXTERNAL_API_URL);
+        
+        this.beanTransformer = new BeanTransformer(baseServletUrl);
+        this.authStoreService = authService.getAuthStoreService();
+        this.pathPattern = getPathRegex();
+    }
+
+    @Override
+    public HttpServletResponse handleDeleteRequest(
+        String pathInfo,
+        QueryParameters queryParams,
+        HttpServletRequest request,
+        HttpServletResponse response
+    ) throws FrameworkException{
+
+        logger.info("handleDeleteRequest() entered");
+
+        String userNumber = extractUserNumberFromUrl(request, pathInfo);
+        IUser user = authStoreService.getUser(userNumber);
+
+        deleteUser(user);
+
+        logger.info("handleDeleteRequest() exiting");
+        return getResponseBuilder().buildResponse(request, response, HttpServletResponse.SC_NO_CONTENT);
+    }
+
+    private String extractUserNumberFromUrl(HttpServletRequest request, String pathInfo) throws InternalServletException{
+
+        try{
```

Comment:
Nitpick - the formatting of this try-catch is a bit off, it would be better like:

```java
try {
...
} catch (Exception ex) {
...
}
```